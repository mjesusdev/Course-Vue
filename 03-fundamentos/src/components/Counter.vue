<template>
  <div>
    <h2>{{ customTitle }}</h2>
    <p> {{ counter }} <sup>2</sup> = {{ squareCounter }} </p>

    <div>
      <button @click="increase">+1</button>
      <button @click="decrease">-1</button>
    </div>

  </div>
</template>

<script>
    export default {
        /* name: 'Counter', */
        props: {
          title: String,
          start: {
            type: Number,
            default: 100,
            validator( value ) {
              return value > 100
            }
            /* required: true, */
          }
        },

        data() {
            return {
              counter: this.start
            }
        },
        methods: {
            getSquareValue() {
              return this.counter * this.counter;
            },

            increase () {
              this.counter++;
            },

            decrease () {
              this.counter--;
            }
        },
        computed: {
            squareCounter() {
              return this.counter * this.counter;
            },

            customTitle() {
              /* return this.title ? this.title : 'Salieron'; */
              return this.title || 'Counter';
            }
        }
    }
</script>

<style>
  button {
    background-color: #64bb87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
  }
</style>