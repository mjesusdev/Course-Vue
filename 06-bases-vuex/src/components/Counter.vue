<template>
  <div>
    <h1>Counter - Vuex</h1>
    <h2>Direct Access: {{ $store.state.count }} </h2>
    <h2>Computed: {{ countComputed }}</h2>

    <button @click="increment">+1</button>
    <button @click="incrementBy">+5</button>
    <button @click="randomInt" :disabled="isLoading">Random</button>

    <h1>mapState</h1>
    <h2>mapState: {{ count }} </h2>
    <h2>LastMutation: {{ lastMutation }}</h2>
  </div>
</template>

<script>

    import { mapState, mapActions } from 'vuex'

    export default {
        computed: {
            countComputed() {
                return this.$store.state.count
            },
            ...mapState(['count', 'lastMutation', 'isLoading'])
            /* ...mapState({
                count: state => state.count,
                lastMutation: state => state.lastMutation
            }) */
        },

        methods: {
            increment() {
                this.$store.commit('increment')
            },
            incrementBy() {
                this.$store.commit('incrementBy', 5)
            },
            /* ...mapActions(['incrementRandomInt']) */
            ...mapActions({
                randomInt: 'incrementRandomInt'
            })
        }
    }
</script>

<style>

</style>