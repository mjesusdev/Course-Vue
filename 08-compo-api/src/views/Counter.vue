<template>
    <div>
        <h1 class="text-success">Counter View</h1>
        <h2 class="">{{ counter }}</h2>
        <button class="btn btn-success m-1" @click="increase">+1</button>
        <button class="btn btn-success m-1" @click="decrease">-1</button>
    </div>
</template>

<script>
    import useCounter from '@/composables/useCounter'

    export default {
        name: 'Counter',
        props: {},
        emits: [],
        setup() {
            const { counter, increase, decrease } = useCounter(25)
            return {
                counter, increase, decrease
            }
        }
    }
</script>

<style>

</style>