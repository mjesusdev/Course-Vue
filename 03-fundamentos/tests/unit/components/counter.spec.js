import Counter from '@/components/Counter'
import { shallowMount, mount } from '@vue/test-utils'

describe('Counter Component', () => {

/*     test('should match with Snapshot', () => {
        const wrapper = shallowMount( Counter )
        expect( wrapper.html() ).toMatchSnapshot()
    }) */

    test('h2 should have the value by default "Counter"', () => {
        const wrapper = shallowMount( Counter )
    
        /* expect( wrapper.find('h10').exists() ).toBeTruthy() */

        const h2Value = wrapper.find('h2').text()

        expect( h2Value ).toBe('Counter')
    })

})