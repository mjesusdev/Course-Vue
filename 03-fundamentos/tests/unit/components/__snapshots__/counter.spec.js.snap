// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component should match with Snapshot 1`] = `
<div>
  <h2>Counter</h2>
  <p>100 <sup>2</sup> = 10000</p>
  <div><button>+1</button><button>-1</button></div>
</div>
`;
